File: C5_PeerCast/PeerCastVB/PeerCast/ThreadMessage.vb
Imports Microsoft.VisualBasic
Imports System
Namespace PeerCast
	Public Class ThreadMessage
		Private privateMessageType As UiMessage
		Public Property MessageType() As UiMessage
			Get
				Return privateMessageType
			End Get
			Set(ByVal value As UiMessage)
				privateMessageType = value
			End Set
		End Property
		Private privateMessage As String
		Public Property Message() As String
			Get
				Return privateMessage
			End Get
			Set(ByVal value As String)
				privateMessage = value
			End Set
		End Property

		Public Shared Function Create(ByVal messageType As UiMessage, ByVal message As String) As ThreadMessage
			Return New ThreadMessage() With {.MessageType = messageType, .Message = message}
		End Function
	End Class
End Namespace

File: C05_PeerCast/PeerCastCS/PeerCast/ThreadMessage.cs
namespace PeerCast
{
    public class ThreadMessage
    {
        public UiMessage MessageType { get; set; }
        public string Message { get; set; }

        public static ThreadMessage Create(UiMessage messageType, string message)
        {
            return new ThreadMessage() { MessageType = messageType, Message = message }; 
        }
    }
}
